toolchain/sem_ir/inst_fingerprinter.cpp

```diff
@@ -243,6 +243,16 @@ struct Worklist {
     AddEntity(sem_ir->named_constraints().Get(named_constraint_id));
   }
 
+  auto Add(RequireImplsId require_id) -> void {
+    if (!require_id.has_value()) {
+      AddInvalid();
+      return;
+    }
```

Comment:
This seems a little unusual, when is it the case that you'd fingerprint a `RequireDecl` without an id associated?

i.e., wondering why this isn't:

```suggestion
    CARBON_CHECK(require_id.has_value());
```

Reply:
It just makes it flexible so it doesn't have to be changed later. There's nothing _bad_ about a None being formatted in the future. But it can be CHECK for now, yeah.